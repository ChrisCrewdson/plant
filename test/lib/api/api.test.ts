export {}; // To get around: Cannot redeclare block-scoped variable .ts(2451)

const helper = require('../../helper');

describe('api', () => {
  beforeAll(async () => {
    const data = await helper.startServerAuthenticated();
    const { userId, port } = data;
    expect(userId).toBeTruthy();
    expect(port).toBeGreaterThan(3000);
  });

  afterAll(() => helper.stopServer());

  test('should get a 404 if the path is not recognized', async () => {
    const reqOptions = {
      method: 'GET',
      authenticate: false,
      text: true,
      url: '/unknown',
    };

    const { httpMsg, response } = await helper.makeRequest(reqOptions);
    expect(response.status).toBe(404);
    expect(httpMsg).toMatchSnapshot();
  });
});
