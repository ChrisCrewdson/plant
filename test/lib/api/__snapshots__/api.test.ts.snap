// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api should get a 404 if the path is not recognized 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <link rel=\\"stylesheet\\" href=\\"https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap\\" />
    <link rel=\\"stylesheet\\" href=\\"https://fonts.googleapis.com/icon?family=Material+Icons\\" />
    
    <link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/apple-touch-icon.png\\">
    <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/favicon-16x16.png\\" sizes=\\"16x16\\">
    <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/favicon-32x32.png\\" sizes=\\"32x32\\">
    <link rel=\\"manifest\\" href=\\"/manifest.json\\">
    <link rel=\\"mask-icon\\" href=\\"/safari-pinned-tab.svg\\" color=\\"#5bbad5\\">
    <meta charset=\\"UTF-8\\">
    <meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\">
    <meta name=\\"theme-color\\" content=\\"#ffffff\\">
    <meta name=\\"viewport\\" content=\\"minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no\\">
    <title>Plaaant</title>
  </head>
<body><div id='wrapper'></div><script>window.__SSR__ = false;
window.__INITIAL_STATE__ = {}</script><script src=\\"/bundle.js\\"></script></body>
</html>"
`;
